### PR TITLE
Add document about registry configuration

### PR DESCRIPTION
**Pull request by miaoyq**
Addresses https://github.com/containerd/cri-containerd/issues/559#issuecomment-370100572; adds documentation describing how to configuration image registry for CRI-Containerd.

/cc @Random-Liu @abhi 

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

File: docs/registry.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Configure Image Registry for CRI-containerd
+This document describes the methods how to configure the image registry for CRI-containerd.
```
Comment (Random-Liu) [inline docs/registry.md]: remove `how`.

File: docs/registry.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+In CRI-containerd, `docker.io` is the default image registry. You can also set  up other image registries similar to docker. CRI-containerd provides two ways to set up the image registry.
```
Comment (Random-Liu) [inline docs/registry.md]: 2 spaces between "set" and "up".

Comment (Random-Liu) [inline docs/registry.md]: Let's use `cri-containerd` in the text, the same with following ones.

Comment (Random-Liu) [inline docs/registry.md]: Add TODO to remove the document for standalone mode and also add `(containerd/containerd#2120)`.

File: docs/registry.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+## Configure Image Registry with `--registry` Option
+
+Start CRI-containerd with `--registry` option like:
+```bash
+$ cri-containerd \
+    --registry=test.secure-registry.io=https://HostIP1:Port1 \
+    --registry=test.insecure-registry.io=http://HostIP2:Port2
+```
+
+If you want configure a insecure registry, you need set the endpoint with `http://` prefix. If you want configure a secure registry, you need set the endpoint with `https://` prefix.
```
Comment (Random-Liu) [inline docs/registry.md]: s/want/want to

File: docs/registry.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+This method is only suitable for standalone mode.
+
```
Comment (Random-Liu) [inline docs/registry.md]: nit: one empty line is enough.

File: docs/registry.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+## Configure Image Registry with Config File in Standalone Mode
+In standalone mode, CRI-containerd also specifies registris using the configuration file located in `/etc/cri-containerd/config.toml`. You can set registries in the configuration file as follows:
+```toml
+#... ...
+[registry]
+  [registry.mirrors]
+    [registry.mirrors."docker.io"]
```
Comment (Random-Liu) [inline docs/registry.md]: Only this is ok for toml.
```
[registry.mirrors."docker.io"]
  endpoint = ["https://registry-1.docker.io"]
```

File: docs/registry.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      endpoint = ["https://registry-1.docker.io"]
+    [registry.mirrors."test.secure-registry.io"]
+      endpoint = ["https://HostIP1:Port1"]
+    [registry.mirrors."test.insecure-registry.io"]
+      endpoint = ["http://HostIP2:Port2"]
+```
+
+In the same way, the default configuration also can be generated by `cri-containerd default-config > /etc/cri-containerd/config.toml`. The endpoint is a list that can contain multiple image registry URLs splited by commas.
```
Comment (Random-Liu) [inline docs/registry.md]: The URLs will be tried one by one from the first to the last until a working one is found. cri-containerd won't be able to pull image if no working URL is found.

File: docs/registry.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+After modify the config file, you need restart the CRI-containerd service.
+
+## Configure Image Registry with Config File in CRI Plugin Mode
+Today, CRI-containerd can also be used as a native plugin of Containerd. In this mode, you should set registries in `/etc/containerd/config.toml` as follows:
```
Comment (Random-Liu) [inline docs/registry.md]: s/Containerd/containerd.

File: docs/registry.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+```toml
+#... ...
+[plugins]
```
Comment (Random-Liu) [inline docs/registry.md]: Also, don't need so many levels.

File: docs/registry.md (hunk 6 above)
Comment (Random-Liu) [inline docs/registry.md]: Move `The endpoint is a list that can contain multiple image registry URLs splited by commas.` up to a separate paragraph.

**Comment by miaoyq** [comment]
@Random-Liu  Thanks for review.
Done.

**Pull request merged**